core/targets.py

```diff
@@ -61,7 +61,7 @@ def target_type(target):
 
 
 def analysis(targets, check_ranges, check_subdomains, subs_temp, range_temp, log_in_file, time_sleep,
-             language, verbose_level, retries, socks_proxy, enumerate_flag):
+             language, verbose_level, retries, socks_proxy, enumerate_flag, scan_method):
```

Comment:
do not modify core for one module

Reply:
Sir here's the problem:
The current tool working need -i for the target and in shodan scan case (CIDR) the target will be : ex: 10.0.0.0/8.
but the internal working of tool converts this CIDR to IP. But thanks to shodan that it supports CIDR cases too with the following api call:
**https://api.shodan.io/shodan/host/search?key=shodan_api_key&query="net: 10.0.0.0/8"**
So if we use the --method-args for CIDR like:
**python nettacker.py -i pintasuper.com -m shodan_scan --method-args "shodan_api_key=<apikey>&shodan_cidr=10.0.0.0/8"**
Then it makes no sense to use -i and giving domain name as we are not using that. But without -i the tool won't work. Well I will fix it. Just need some time. will try to resolve it ASAP.

Reply:
I think it should be like this:
**python nettacker.py -i pintasuper.com -m shodan_scan --method-args "shodan_api_key=XXXYYYYY&shodan_query_override=net:10.0.0.0/8"**

in this case `shodan_query_override` would be used and the target would be ignored.

if `shodan_query_override` method arg is defined it will allow the module to run some cool Shodan queries like this:

**python nettacker.py -i dummy.com -m  shodan_scan --method-args "shodan_api_key=XXXYYYY&shodan_query_override=country:in+port:6443"**  - this will return all instances of Kubernetes running in India (no way to define this using normal nettacker syntax"


Don't forget that Nettacker allows users to chain modules, so for example a user might want to do a subdomain scan and a shodan scan at the same time  on a target:

**python nettacker.py -i pintasuper.com -m subdomain_scan,shodan_scan --method-args "shodan_api_key=XXXYYYYY"** - in this case both scan methods would run

Reply:
This is great idea. Actually it also give user simple user interface and he can type his own query. We need not to mention for CIDR now. The query which he types will automatically run and nettacker will give results according to it :)

Reply:
One doubt. If user not gives that shodan_query_override then what should be the default query that should be run?
It is normal hostname scan?